src/utils.jl

```diff
@@ -415,3 +415,63 @@ macro hijack_observable(name)
     end)
 
 end
+
+############################################################
+#                                                          #
+#     Splitting Of LineString / coast lines                #
+#                                                          #
+############################################################
+
+#`LineSplitting` module defines a `split` method for vectors of `LineString` objects.
+#
+#This is needed to fix e.g. coast line displays when lon_0 is not 0 but cutting polygons at lon_0+-180.
+
+module LineSplitting
+
+	import GeoMakie.LineString
+	import GeoMakie.Observable
+	import Base.split
+	
+	function regroup(tmp::Vector)
+		coastlines_custom=LineString[]
+		println(typeof(coastlines_custom))
+		for ii in 1:length(tmp)
+			push!(coastlines_custom,tmp[ii][:]...)
+		end
+		coastlines_custom
+	end
+	
+	function split(tmp::Vector,lon0=-160.0)
+		[split(a,lon0) for a in tmp]
+	end
+	
+	function split(tmp::LineString,lon0=-160.0)
+		lon0<0.0 ? lon1=lon0+180 : lon1=lon0-180 
+		np=length(tmp)
+		tmp2=fill(0,np)
+		for p in 1:np
+			tmp1=tmp[p]
+			tmp2[p]=maximum( [(tmp1[1][1]<=lon1)+2*(tmp1[2][1]>=lon1) , (tmp1[2][1]<=lon1)+2*(tmp1[1][1]>=lon1)] )
+		end
+		if sum(tmp2.==3)==0
+			[tmp]
+		else
+			jj=[0;findall(tmp2.==3)...;np+1]
+			[LineString(tmp[jj[ii]+1:jj[ii+1]-1]) for ii in 1:length(jj)-1]
+		end
+	end
```

Comment:
I think this can be improved syntactically (if I understand your code correctly), by:
```suggestion
        getlon(p::GeometryBasics.Point) = p[1]
        
	function split(tmp::LineString,lon0=-160.0)
		lon1 = lon0 + (lon0 < 0 ? 180 : -180)
		linenodes = GeometryBasics.coordinates(tmp)  # get coordinates of line nodes
		# Find nodes that are on either side of lon0
		cond = getlon.(linenodes) .>= lon0
		# Find interval starts and ends
		end_cond = diff(cond)  # nonzero values denote ends of intervals
		end_inds = findall(!=(0), end_cond)
		start_inds = [firstindex(linenodes);  end_inds .+ 1]  # starts of intervals
		end_inds = [end_inds; lastindex(linenodes)]  # ends of intervals
		# do the splitting
		split_coords = view.(Ref(linenodes), UnitRange.(start_inds, end_inds))  # For each start-end pair, get those coords
		# reconstruct lines from points
		split_lines = GeometryBasics.LineString.(split_coords) 
	end
```

My expectation is that concatenating all the split nodes should be the same as the original nodes, which this suggestion satisfies, i.e.
```julia
vcat(split_coords...) == linenodes
```

In your code however, it seems that if the input is 
```julia
julia> tmp = GeoMakie.coastlines()[1]  # longitudes of ≈ - 161 ± 2

julia> lon0 = 20  # i.e. I want to split the coastline along -160

julia> split_lines = split(tmp, lon0)

julia> vcat(coordinates.(split_lines))
16-element Vector{GeometryBasics.Point{2, Float32}}:
 [-163.71289, -78.595665]
 [-163.1058, -78.223335]
 [-163.1058, -78.223335]
 [-161.24512, -78.38018]
 [-161.24512, -78.38018]
 [-160.2462, -78.69365]
 [-159.4824, -79.04634]
 [-159.20819, -79.49701]
 [-161.1276, -79.63421]
 [-162.43985, -79.28146]
 [-162.43985, -79.28146]
 [-163.0274, -78.92877]
 [-163.0274, -78.92877]
 [-163.0666, -78.869965]
 [-163.0666, -78.869965]
 [-163.71289, -78.595665]
```
then there are many duplicate coordinates in the output.

Reply:
Thank you very much for the suggested improvements. I tested your revised code back in Nov and it worked well from what I remember. Wanted to take another look to confirm I understood your code but then I got side tracked -- apologies for that.

Issue now though is that my basic test case no longer works (`ax = GeoMakie.GeoAxis(f[1,1]; dest = "+proj=longlat +datum=WGS84 +lon_0=-160",lonlims=GeoMakie.automatic)`). I assume it's because of code merged ahead of this one but I am not sure.

Reply:
Have now merged master into branch and updated example code, so I could try suggested change again. Might wait to hear from  @asinghvi17 though